lib/cli/run-helpers.js

```diff
@@ -0,0 +1,335 @@
+
+  // flush output for Node.js Windows pipe bug
+  // https://github.com/joyent/node/issues/6247 is just one bug example
+  // https://github.com/visionmedia/mocha/issues/333 has a good discussion
+  const done = () => {
+    if (!draining--) {
+      process.exit(clampedCode);
+    }
+  };
+
+  const streams = [process.stdout, process.stderr];
+
+  streams.forEach(stream => {
+    // submit empty write request and wait for completion
+    draining += 1;
+    stream.write('', done);
+  });
+
+  done();
+};
+
+/**
+ * Hide the cursor.
+ * @private
+ */
+const hideCursor = () => {
+  process.stdout.write('\u001b[?25l');
+};
+
+/**
+ * Show the cursor.
+ * @private
+ */
+const showCursor = () => {
+  process.stdout.write('\u001b[?25h');
+};
+
+/**
+ * Stop cursor business
+ * @private
+ */
+const stop = () => {
+  process.stdout.write('\u001b[2K');
+};
+
+/**
+ * Dumps a sorted list of the enumerable, lower-case keys of some object
+ * to `STDOUT`.
+ * @param {Object} obj - Object, ostensibly having some enumerable keys
+ * @private
+ */
+const showKeys = obj => {
+  console.log();
+  Object.keys(obj)
+    .filter(value => /^[a-z]/.test(value))
+    .sort()
+    .forEach(key => {
+      console.log(`    ${key}`);
+    });
+  console.log();
+};
+
+/**
+ * Coerce a comma-delimited string (or array thereof) into a flattened array of
+ * strings
+ * @param {string|string[]} str - Value to coerce
+ * @returns {string[]} Array of strings
+ * @private
+ */
+exports.list = str =>
+  Array.isArray(str) ? exports.list(str.join(',')) : (str || '').split(/ *, */);
+
+/**
+ * Dump list of built-in interfaces
+ * @private
+ */
+exports.runShowInterfaces = () => {
+  showKeys(Mocha.interfaces);
+};
+
+/**
+ * Dump list of built-in reporters
+ * @private
+ */
+exports.runShowReporters = () => {
+  showKeys(Mocha.reporters);
+};
+
+/**
+ * Process extensions for `--watch`, `--compilers`, etc.
+ * @param {yargs.Arguments} argv - Arguments
+ * @private
+ */
+exports.handleExtensions = argv => {
+  const compilers = argv.compilers || [];
+  if (compilers.length) {
+    require('util').deprecate(() => {},
+    '"--compilers" will be removed in a future version of Mocha; see https://git.io/vdcSr for more info')();
+  }
+
+  argv.extensions = ['js'];
+  compilers.forEach(c => {
+    const idx = c.indexOf(':');
+    const ext = c.slice(0, idx);
+    let mod = c.slice(idx + 1);
+
+    if (mod[0] === '.') {
+      mod = path.join(cwd, mod);
+    }
+    require(mod);
+    argv.extensions.push(ext);
+    argv.watchExtensions.push(ext);
+  });
+};
+
+/**
+ * `require()` the modules as required by `--require <require>`
+ * @param {yargs.Arguments} argv - Arguments
+ * @private
+ */
+exports.handleRequires = argv => {
+  argv.require.forEach(mod => {
+    let modpath = mod;
+    if (fs.existsSync(mod, {cwd}) || fs.existsSync(`${mod}.js`, {cwd})) {
+      modpath = path.resolve(mod);
+      debug(`resolved ${mod} to ${modpath}`);
+    }
+    require(modpath);
+  });
+};
```

Comment:
For my money, I'd prefer these `handleXXX` methods **not** be written in terms of `yargs` implementation.

```js
/*
 *  @param {string[]} modules - Modules to be `require`d.
 */
exports.handleRequires = modules => {
  modules.forEach(mod => {
```
then
```js
  handleRequires(argv.require);
```

Reply:
ah, hmm.  I don't feel strongly about it either way, ok.

- [ ] don't pass `argv` around here

Reply:
did the best I could with this

Reply:
Easier to test in isolation.